test/functional/tests/security/test_security_user.py

```diff
@@ -0,0 +1,474 @@
+from core.test_run import TestRun
+from test_tools import fs_utils
+from test_tools.disk_utils import Filesystem
+from test_utils.output import CmdException
+from test_utils.size import Size, Unit
+
+mount_point = "/mnt/cas"
+system_casadm_bin_path = "/sbin/casadm"
+user_casadm_bin_dest_path = "/bin/casadm"
+ioclass_config_path = "/etc/opencas/ioclass-config.csv"
+ioclass_config_copy_path = "/etc/opencas/ioclass-config-copy.csv"
+user_name = "user"
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_user_cli():
+    """
+        title: Test that OpenCAS do not allow to change parameters in CLI by non-root user.
+        description: |
+          Checking if changing parameters in CLI by non-root user is forbidden by OpenCAS,
+          but is permitted with 'sudo' command.
+        pass_criteria:
+          - Non-root user can only print help and CAS version.
+          - Sudoer user is allowed to change OpenCAS parameters in CLI with sudo.
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+
+    with TestRun.step("Start OpenCAS."):
+        cache = casadm.start_cache(cache_dev, force=True)
+
+    with TestRun.step("Add core to cache and mount it."):
+        core_dev.create_filesystem(Filesystem.ext3)
+        core = cache.add_core(core_dev)
+        core.mount(mount_point)
+
+    with TestRun.step(f"Copy casadm bin from {system_casadm_bin_path} "
+                      f"to {user_casadm_bin_dest_path}."):
+        casadm_bin = fs_utils.parse_ls_output(fs_utils.ls_item(f"{system_casadm_bin_path}"))[0]
+        casadm_bin_copy = casadm_bin.copy(user_casadm_bin_dest_path, True)
+        casadm_bin_copy.chmod_numerical(777)
+
+    with TestRun.step("Copy IO class config."):
+        io_conf = fs_utils.parse_ls_output(fs_utils.ls_item(f"{ioclass_config_path}"))[0]
+        io_conf_copy = io_conf.copy(ioclass_config_copy_path, force=True)
+
+    with TestRun.step("Unmount core."):
+        core.unmount()
+
+    with TestRun.step("Stop OpenCAS."):
+        casadm.stop_all_caches()
+
+    with TestRun.step("Add non-root user account."):
+        TestRun.executor.run(f"useradd -N -r -l {user_name}")
+        user_home_dir = fs_utils.parse_ls_output(fs_utils.ls_item(f"/home/{user_name}"))[0]
+        user_home_dir.chmod_numerical(777, True)
+
+    with TestRun.step("Try to start OpenCAS."):
+        try:
+            output = run_as_other_user(cli.start_cmd(cache_dev.system_path), user_name)
+            if output.exit_code == 0:
+                TestRun.fail("Starting OpenCAS should fail!")
```

Comment:
cache
In this test we want to check all commands, whether they pass or fail - TestRun.fail will stop test, TestRun.LOGGER.error will set result to fail, but continue with the test